spec/src/main/asciidoc/repository.asciidoc

```diff
@@ -123,7 +124,222 @@ In Jakarta Data, an entity refers to a fundamental data representation and manag
+    },
+    {
+      "title": "Sample Book 2",
+      "category": "Non-fiction"
+    }
+  ]
+}
+----
+
+In the JSON representation, an `Author` entity can be associated with multiple `Book` entities within an array.
+
+[options="header"]
+|===
+| Author Table | Book Table | Author_Book (Auxiliary) Table
+
+| id (Primary Key) | id (Primary Key) | author_id (Foreign Key)
+| name | title | book_id (Foreign Key)
+
+|===
+
+In a relational database, this relationship typically generates three tables: `Author`, `Book`, and an auxiliary table `Author_Book` to manage the many-to-many relationship between authors and books.
+
+In some scenarios, books can have multiple authors, and authors can contribute to several books, resulting in a many-to-many (N-N) cardinality relationship. Jakarta Data offers flexibility in representing and managing such complex relationships.
+
+This N-N relationship typically generates a dedicated table to manage the associations between books and authors in a relational database. However, NoSQL databases may take a different approach, especially in cases where denormalization and data duplication are favored for query-driven designs.
+
+IMPORTANT: The key-value database might support these fields, generating a single BLOB value. However, the serialization process for such fields may vary depending on the Jakarta Data provider and the specific NoSQL database used.
+
+NOTE: The Graph database is not required to support this type, but it might be used for aggregate query returns or as a read-only field.
+
+===== Recursion at Domain-Relation Fields in Jakarta Data
+
+In the context of Jakarta Data, the term "recursion" pertains to the ability to manage hierarchical or nested relationships between entities. This capability is essential when dealing with complex domain models involving associations, aggregations, or compositions between various entity types. Jakarta Data ensures these relationships are correctly mapped and maintained within the database, enabling consistent data retrieval and manipulation.
+
+
+Within relational databases, Jakarta Data necessitates support for recursive relationships. It implies that if one entity type is associated with or contains another entity type, the Jakarta Data provider for relational databases must establish the required table structures and foreign key constraints to uphold these relationships. This approach guarantees data integrity and consistency when working with the database.
+
+On the other hand, NoSQL databases, known for their flexible and schema-less nature, may not require explicit support for recursive relationships as dictated by the Jakarta Data specification. In NoSQL databases, data is often stored in a denormalized or nested document format, making it more challenging to enforce strict hierarchical relationships. Instead, NoSQL databases may emphasize query-driven design and may not necessitate explicit mapping or management of recursive relationships.
```

Comment:
```suggestion
For other types of databases, Jakarta Data does not require explicit support for recursive relationships.  In NoSQL databases, data is often stored in a denormalized or nested document format, making it more challenging to enforce strict hierarchical relationships. Instead, NoSQL databases may emphasize query-driven design rather than explicit mapping or management of recursive relationships.
```